pennylane/devices/task_qubit.py

```diff
@@ -0,0 +1,372 @@
+
+    >>> if __name__ == '__main__': # Required for LocalCluster
+    ...     cluster = dist.LocalCluster(n_workers=4, threads_per_worker=1)
+    ...     client = dist.Client(cluster)
+    ...     backend = "default.qubit"
+    ...     dev = qml.device("task.qubit", wires=6, backend=backend)
+    ...     @qml.qnode(dev, cache=False, interface="tf", diff_method="parameter-shift") # caching must be disabled due to proxy interface
+    ...     def circuit(x):
+    ...         qml.RX(x[0], wires=0)
+    ...         qml.RY(x[1], wires=0)
+    ...         qml.RZ(x[2], wires=0)
+    ...         qml.RZ(x[0], wires=1)
+    ...         qml.RX(x[1], wires=1)
+    ...         qml.RY(x[2], wires=1)
+    ...         return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+    ...     weights = tf.Variable(np.random.rand(3))
+    ...     def f_submit(weights):
+    ...         with tf.GradientTape() as tape:
+    ...             # Use the circuit to calculate the loss value
+    ...             loss = tf.abs(circuit(weights)-0.5)**2
+    ...         return tape.gradient(loss, weights)
+    ...     print(client.submit(f_submit, weights).result())
+    tf.Tensor([0.01776833 0.05199685 0.03689981], shape=(3,), dtype=float64)
+
+    For self-encapsulated workflows, the task-based ``qml.taskify`` command can also be employed, which will allow automatic offloading using the
+    batch execute support in PennyLane devices. As an example:
+
+    >>> def my_workflow(params, backend, interface, diff_method):
+    ...     qpu = qml.device(
+    ...         "task.qubit",
+    ...         wires=3,
+    ...         backend=backend
+    ...     )
+    ...     # Caching must be disabled, as we are using a proxy device
+    ...     @qml.qnode(qpu, cache=False, interface=interface, diff_method=diff_method)
+    ...     def circuit(x):
+    ...         qml.RX(x[0], wires=0)
+    ...         qml.RY(x[0], wires=0)
+    ...         qml.RZ(x[1], wires=0)
+    ...         qml.RX(x[1], wires=1)
+    ...         qml.RY(x[2], wires=1)
+    ...         qml.RZ(x[2], wires=1)
+    ...         return [qml.expval(qml.PauliZ(i) @ qml.PauliZ((i+1)%3)) for i in range(3)]
+    ...
+    ...     # Need a local copy of data on device
+    ...     if interface == "tf":
+    ...         weights = tf.Variable(params)
+    ...         with tf.GradientTape() as tape:
+    ...             # Use the circuit to calculate the loss value
+    ...             loss = circuit(weights)
+    ...         w_grad = tape.jacobian(loss, [weights])
+    ...     elif interface == "torch":
+    ...         weights = torch.tensor(params, requires_grad=True)
+    ...         w_grad = torch.autograd.functional.jacobian(circuit, weights)
+    ...     else:
+    ...         weights = qml.numpy.array(params, requires_grad=True)
+    ...         w_grad = qml.jacobian(circuit)(weights)
+    ...     return w_grad
+
+    >>> qml.taskify(my_workflow)(np.array([1.0,2.0,3.0]), "default.qubit", "autograd", "backprop")
```

Comment:
This is used to add a callable to the worker. If you have access to a dask cluster external to your machine (and a client locally that is connected to the external cluster), it submits the callable function and the supplied parameters to run there.

It can be thought of similar to having a login node and worker node on a cluster: the submission happens on the login node level, and offloads the computation to a provisioned worker node.